src/rebalancer/bridges/LZUnifiedBridge.sol

```diff
@@ -0,0 +1,264 @@
+            msgReceipt.guid
+        );
+    }
+
+    function lzCompose(
+        address from,
+        bytes32 /*guid*/,
+        bytes calldata message,
+        address /*executor*/,
+        bytes calldata /*extraData*/
+    ) external payable {
+        if (msg.sender != endpoint) revert LZBridge_OnlyEndpoint();
+
+        (address market) = abi.decode(message, (address));
+        address underlying = ImTokenMinimal(market).underlying();
+
+        address bridgeContract = bridgeContracts[underlying];
+        if (bridgeContract == address(0)) {
+            bridgeContract = underlying;
+        }
+
+        if (from != bridgeContract) revert LZBridge_BadFrom();
+
+        address executor = oftExecutors[underlying];
+        if (executor == address(0)) revert LZBridge_ExecutorNotSet();
+
+        bytes memory data = abi.encodeWithSelector(
+            IOftMessageExecutor.executeCompose.selector,
+            market,
+            underlying,
+            bridgeContract
+        );
+
+        (bool ok, bytes memory ret) = executor.delegatecall(data);
+        if (!ok) {
+            assembly {
+                revert(add(ret, 32), mload(ret))
+            }
+        }
+    }
+
+    function processUncomposedMessages(address _market)
+        external
+        payable
+        onlyBridgeConfigurator
+    {
+        address underlying = ImTokenMinimal(_market).underlying();
+        address bridgeContract = bridgeContracts[underlying];
+        if (bridgeContract == address(0)) {
+            bridgeContract = underlying;
+        }
+
+        address executor = oftExecutors[underlying];
+        if (executor == address(0)) revert LZBridge_ExecutorNotSet();
+
+        bytes memory data = abi.encodeWithSelector(
+            IOftMessageExecutor.processUncomposed.selector,
+            _market,
+            underlying,
+            bridgeContract
+        );
+
+        (bool ok, bytes memory ret) = executor.delegatecall(data);
+        if (!ok) {
+            assembly {
+                revert(add(ret, 32), mload(ret))
+            }
+        }
+    }
+
+    // ----------- PRIVATE ------------
+    function _getFee(uint32 dstEid, bytes memory _message)
+        private
+        view
+        returns (MessagingFee memory fees, SendParam memory lzSendParams)
+    {
+        (address market, uint256 amountLD, uint256 minAmountLD, bytes memory extraOptions) =
+            abi.decode(_message, (address, uint256, uint256, bytes));
+
+        address _underlying = ImTokenMinimal(market).underlying();
+        address _bridgeContract = bridgeContracts[_underlying];
+        if (_bridgeContract == address(0)) {
+            _bridgeContract = _underlying;
+        }
+
+        lzSendParams = SendParam({
+            dstEid: dstEid,
+            to: bytes32(uint256(uint160(market))), 
+            amountLD: amountLD,
+            minAmountLD: minAmountLD,
+            extraOptions: extraOptions,
+            composeMsg: "",
+            oftCmd: ""
```

Comment:
`composeMsg` is empty and `to` is the market, so `lzCompose` never fires and the executor never unwraps. Because of this, markets receive OFT/adapter tokens directly. 

If I understand the implementation correctly, we need to route to the bridge and set a compose payload, so that we achieve the desired behavior.